contracts/test/DelegateE2E.t.sol

```diff
@@ -369,6 +376,133 @@ contract TestIntegrationJBXBuybackDelegate is Test, UniswapV3ForgeQuoter {
     assertEq(jbController.reservedTokenBalanceOf(1), _reservedBalanceBefore);
   }
 
+    /**
+   * @notice If the amount of token returned by swapping is greater than by minting, swap & use quote from uniswap lib
+   *
+   * @dev    Should swap for both beneficiary and reserve (by burning/minting)
+   */
+  function test_swapWhenQuoteNotProvidedInMetadata(uint256 _amountIn) public {
+    _amountIn = bound(_amountIn, 100, 100 ether);
+  
+    // Reconfigure with a weight of 1  
+    _reconfigure(1, address(delegate), 1, 0);
+
+    uint256 _reservedBalanceBefore = jbController.reservedTokenBalanceOf(1);
+
+    uint256 _quote = getAmountOut(pool, _amountIn, address(weth));
```

Comment:
I think it would be usefull to get the twap too (the quoter only return the spot price), via uni oracle lib I guess